File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.cpp
class Solution {
public:
    long long maximumSubarraySum(vector<int>& nums, int k) {
        unordered_set<int> us;
        int n = nums.size();
        long long s = 0;
        int j = 0;
        long long maxEle = 0;
        for (int i = 0; i < n; ++i) {
            while (j < n and us.size() and us.find(nums[i]) != us.end()) {
                us.erase(nums[j]);
                s -= nums[j++];
            }
            s += 1LL * nums[i];
            us.insert(nums[i]);
            if (k < us.size()) {
                us.erase(nums[j]);
                s -= nums[j++];
            }
            cout << s << '\n';
            if (us.size() == k) {
                maxEle = max(maxEle, s);
            }
        }
        return maxEle;
    }
};

File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.py
class Solution:
    def maximumSubarraySum(self, nums: List[int], k: int) -> int:
        n = len(nums)
        j = 0
        unique = set()
        total = 0
        ans = 0
        
        for i in range(n):
            while j < n and len(unique) > 0 and nums[i] in unique:
                unique.discard(nums[j])
                total -= nums[j]
                j += 1
                
            total += nums[i]
            unique.add(nums[i])            
                
            if k == len(unique):
                ans = max(ans, total)
                total -= nums[j]
                unique.discard(nums[j])
                j += 1
                
        return ans